src/RpcServer/RpcServer.Blockchain.cs

```diff
@@ -103,11 +103,24 @@ protected virtual JObject GetBlockHeader(JArray _params)
         [RpcMethod]
         protected virtual JObject GetContractState(JArray _params)
         {
-            UInt160 script_hash = UInt160.Parse(_params[0].AsString());
+            UInt160 script_hash = UInt160.Parse(ToScriptHash(_params[0].AsString()));
             ContractState contract = Blockchain.Singleton.View.Contracts.TryGet(script_hash);
             return contract?.ToJson() ?? throw new RpcException(-100, "Unknown contract");
         }
 
+        public static string ToScriptHash(string keyword)
+        {
+            foreach (var native in NativeContract.Contracts)
+            {
+                if (keyword.Equals(native.Name, StringComparison.InvariantCultureIgnoreCase) ||
```

Comment:
https://github.com/neo-project/neo/blob/a1590b0995b201100b2153d2c0e638732eda0cee/src/neo/SmartContract/Native/NativeContract.cs#L101-L105

You can get native contract by name directly.

Reply:
I referenced this code.
https://github.com/neo-project/neo-modules/blob/master/src/RpcClient/Utility.cs#L39-L50

`NativeContract.GetContract(string name)` This method doesn't support searching by ID.

Reply:
You can use `int.TryParse()` to check whether it is an ID or a name.

Reply:
If it's an ID, you also need to iterate through all NativeContracts. The code doesn't get any simpler.